src/main/java/ifsul/agileproject/rachadinha/controller/UserController.java

```diff
@@ -0,0 +1,75 @@
+package ifsul.agileproject.rachadinha.controller;
+
+import ifsul.agileproject.rachadinha.domain.entity.User;
+import ifsul.agileproject.rachadinha.repository.UserRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+
+    private final UserRepository userRepository;
+
+    public UserController(UserRepository userRepository){
+        this.userRepository = userRepository;
+    }
+
+    //Buscar user por ID
+    @GetMapping("{id}")
+    public User getUserByID(@PathVariable Integer id){
+        return userRepository
```

Comment:
This needs to be a concern handled by a Service class, instead of a repository being accessed directly in the controller.